### PR TITLE
FHAC-1053: Solved Sonarqube blockers and some critical issues in ConnectDirectConfig.

### PR DESCRIPTION
**Pull request by danielrf123**
The DAOUtils class needed to be refactored so that it could use the new HibernateUtil bean.

@TabassumJafri @mhpnguyen please review.

### CODE REVIEW

File: Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/helpers/DaoUtils.java [hunk 1]
```diff
@@ -44,45 +88,49 @@
...
         if (session != null) {
             try {
                 session.close();
             } catch (Exception e) {
-                log.error("Failed to close session: " + e.getMessage(), e);
+                LOG.error("Failed to close session: ", e);
             }
         } else {
-            log.warn("Session is null, cannot close");
+            LOG.warn("Session is null, cannot close");
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/helpers/DaoUtils.java]: Do we need this else block?

File: Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/AddressDaoImpl.java [hunk 2]
```diff
@@ -266,25 +265,27 @@ public Address get(String name) {
...
 
-            if (session != null) {
-                if (names != null && names.size() > 0) {
-                    query = session.getNamedQuery("getAddress");
-
-                    query.setParameterList("emailList", names);
-                } else {
-                    query = session.getNamedQuery("getAddressByStatus");
-                }
+            if (names != null && !names.isEmpty()) {
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/AddressDaoImpl.java]: Can we use CollectionUtils.isNotEmpty from apache commons here?

File: Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/helpers/DaoUtils.java (hunk 1 above)
Comment (danielrf123) [inline Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/helpers/DaoUtils.java]: No need to keep the warning there any longer.  Can be taken out.

File: Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/AddressDaoImpl.java (hunk 2 above)
Comment (danielrf123) [inline Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/AddressDaoImpl.java]: Yes

**Comment by TabassumJafri** [comment]
Endorsing for merge

**Comment by mhpnguyen** [comment]
Endorse for merging without comments

**Pull request merged**